File: measurecompute/DAQ/C#/TimerOutStart01/TimerOutStart01.cs
// ==============================================================================
//
//  File:                         TimerOutStart01.CS
//
//  Library Call Demonstrated:    MccDaq.MccBoard.TimerStart()
//																MccDaq.MccBoard.TimerStop()
//
//  Purpose:                      Controls an Output Timer Channel.
//
//  Demonstration:                Sends a frequency output to Timer 0.
//
//  Other Library Calls:          MccDaq.MccService.ErrHandling()
//
//  Special Requirements:         Board 0 must have a Timer output.
//
// ==============================================================================
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using System.Diagnostics;

using MccDaq;

namespace TimerOutput
{
	public class frmTimerOutput : System.Windows.Forms.Form
	{
		// Required by the Windows Form Designer
		private System.ComponentModel.IContainer components;

		public ToolTip ToolTip1;
		public Button cmdEndProgram;
		public Label lblShowFrequency;
		public Label lblFrequency;
		public Label lblShowValue;
		public Label lblValueSent;
		public Label lblUseScroll;
		public Label lblDemoFunction;
		public Button UpdateButton;

		private MccDaq.MccBoard DaqBoard;
		const int Chan = 0;
		public System.Windows.Forms.TextBox txtFrequencyToSet;

		public frmTimerOutput()
		{     
			// This call is required by the Windows Form Designer.
			InitializeComponent();

			//  Initiate error handling
			//   activating error handling will trap errors like
			//   bad channel numbers and non-configured conditions.
			//   Parameters:
			//     MccDaq.ErrorReporting.PrintAll :all warnings and errors encountered will be printed
			//     MccDaq.ErrorHandling.StopAll   :if an error is encountered, the program will stop
			
			MccDaq.ErrorInfo ULStat = MccDaq.MccService.ErrHandling(MccDaq.ErrorReporting.PrintAll, MccDaq.ErrorHandling.StopAll);
			

			DaqBoard = new MccDaq.MccBoard(0);
		}

		// Form overrides dispose to clean up the component list.
		protected override void  Dispose(bool Disposing)
		{
			if (Disposing)
			{
				if (components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose(Disposing);
		}
 
	    
		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
	    
		private void InitializeComponent()
		{
			this.components = new System.ComponentModel.Container();
			this.ToolTip1 = new System.Windows.Forms.ToolTip(this.components);
			this.cmdEndProgram = new System.Windows.Forms.Button();
			this.txtFrequencyToSet = new System.Windows.Forms.TextBox();
			this.lblShowFrequency = new System.Windows.Forms.Label();
			this.lblFrequency = new System.Windows.Forms.Label();
			this.lblShowValue = new System.Windows.Forms.Label();
			this.lblValueSent = new System.Windows.Forms.Label();
			this.lblUseScroll = new System.Windows.Forms.Label();
			this.lblDemoFunction = new System.Windows.Forms.Label();
			this.UpdateButton = new System.Windows.Forms.Button();
			this.SuspendLayout();
			// 
			// cmdEndProgram
			// 
			this.cmdEndProgram.BackColor = System.Drawing.SystemColors.Control;
			this.cmdEndProgram.Cursor = System.Windows.Forms.Cursors.Default;
			this.cmdEndProgram.Font = new System.Drawing.Font("Arial", 8F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.cmdEndProgram.ForeColor = System.Drawing.SystemColors.ControlText;
			this.cmdEndProgram.Location = new System.Drawing.Point(248, 216);
			this.cmdEndProgram.Name = "cmdEndProgram";
			this.cmdEndProgram.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.cmdEndProgram.Size = new System.Drawing.Size(55, 26);
			this.cmdEndProgram.TabIndex = 5;
			this.cmdEndProgram.Text = "Quit";
			this.cmdEndProgram.Click += new System.EventHandler(this.cmdEndProgram_Click);
			// 
			// txtFrequencyToSet
			// 
			this.txtFrequencyToSet.AcceptsReturn = true;
			this.txtFrequencyToSet.AutoSize = false;
			this.txtFrequencyToSet.BackColor = System.Drawing.SystemColors.Window;
			this.txtFrequencyToSet.BorderStyle = System.Windows.Forms.BorderStyle.FixedSingle;
			this.txtFrequencyToSet.Cursor = System.Windows.Forms.Cursors.IBeam;
			this.txtFrequencyToSet.Font = new System.Drawing.Font("Arial", 8F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.txtFrequencyToSet.ForeColor = System.Drawing.SystemColors.WindowText;
			this.txtFrequencyToSet.Location = new System.Drawing.Point(120, 104);
			this.txtFrequencyToSet.MaxLength = 0;
			this.txtFrequencyToSet.Name = "txtFrequencyToSet";
			this.txtFrequencyToSet.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.txtFrequencyToSet.Size = new System.Drawing.Size(81, 25);
			this.txtFrequencyToSet.TabIndex = 0;
			this.txtFrequencyToSet.Text = "100000";
			// 
			// lblShowFrequency
			// 
			this.lblShowFrequency.BackColor = System.Drawing.SystemColors.Window;
			this.lblShowFrequency.Cursor = System.Windows.Forms.Cursors.Default;
			this.lblShowFrequency.Font = new System.Drawing.Font("Arial", 8F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.lblShowFrequency.ForeColor = System.Drawing.Color.Blue;
			this.lblShowFrequency.Location = new System.Drawing.Point(248, 176);
			this.lblShowFrequency.Name = "lblShowFrequency";
			this.lblShowFrequency.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.lblShowFrequency.Size = new System.Drawing.Size(80, 17);
			this.lblShowFrequency.TabIndex = 6;
			// 
			// lblFrequency
			// 
			this.lblFrequency.BackColor = System.Drawing.SystemColors.Window;
			this.lblFrequency.Cursor = System.Windows.Forms.Cursors.Default;
			this.lblFrequency.Font = new System.Drawing.Font("Arial", 8F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.lblFrequency.ForeColor = System.Drawing.SystemColors.WindowText;
			this.lblFrequency.Location = new System.Drawing.Point(8, 176);
			this.lblFrequency.Name = "lblFrequency";
			this.lblFrequency.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.lblFrequency.Size = new System.Drawing.Size(240, 17);
			this.lblFrequency.TabIndex = 7;
			// 
			// lblShowValue
			// 
			this.lblShowValue.BackColor = System.Drawing.SystemColors.Window;
			this.lblShowValue.Cursor = System.Windows.Forms.Cursors.Default;
			this.lblShowValue.Font = new System.Drawing.Font("Arial", 8F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.lblShowValue.ForeColor = System.Drawing.Color.Blue;
			this.lblShowValue.Location = new System.Drawing.Point(248, 159);
			this.lblShowValue.Name = "lblShowValue";
			this.lblShowValue.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.lblShowValue.Size = new System.Drawing.Size(80, 17);
			this.lblShowValue.TabIndex = 4;
			// 
			// lblValueSent
			// 
			this.lblValueSent.BackColor = System.Drawing.SystemColors.Window;
			this.lblValueSent.Cursor = System.Windows.Forms.Cursors.Default;
			this.lblValueSent.Font = new System.Drawing.Font("Arial", 8F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.lblValueSent.ForeColor = System.Drawing.SystemColors.WindowText;
			this.lblValueSent.Location = new System.Drawing.Point(8, 159);
			this.lblValueSent.Name = "lblValueSent";
			this.lblValueSent.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.lblValueSent.Size = new System.Drawing.Size(240, 17);
			this.lblValueSent.TabIndex = 3;
			// 
			// lblUseScroll
			// 
			this.lblUseScroll.BackColor = System.Drawing.SystemColors.Window;
			this.lblUseScroll.Cursor = System.Windows.Forms.Cursors.Default;
			this.lblUseScroll.Font = new System.Drawing.Font("Arial", 8F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.lblUseScroll.ForeColor = System.Drawing.SystemColors.WindowText;
			this.lblUseScroll.Location = new System.Drawing.Point(56, 64);
			this.lblUseScroll.Name = "lblUseScroll";
			this.lblUseScroll.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.lblUseScroll.Size = new System.Drawing.Size(225, 32);
			this.lblUseScroll.TabIndex = 2;
			this.lblUseScroll.Text = "Enter a frequency within the timer\'s range and click Update Button when done";
			// 
			// lblDemoFunction
			// 
			this.lblDemoFunction.BackColor = System.Drawing.SystemColors.Window;
			this.lblDemoFunction.Cursor = System.Windows.Forms.Cursors.Default;
			this.lblDemoFunction.Font = new System.Drawing.Font("Arial", 8F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.lblDemoFunction.ForeColor = System.Drawing.SystemColors.WindowText;
			this.lblDemoFunction.Location = new System.Drawing.Point(38, 24);
			this.lblDemoFunction.Name = "lblDemoFunction";
			this.lblDemoFunction.RightToLeft = System.Windows.Forms.RightToLeft.No;
			this.lblDemoFunction.Size = new System.Drawing.Size(256, 25);
			this.lblDemoFunction.TabIndex = 1;
			this.lblDemoFunction.Text = "Demonstration of MccDaq.MccBoard.TimerStart() and MccDaq.MccBoard.TimerStop()";
			this.lblDemoFunction.TextAlign = System.Drawing.ContentAlignment.TopCenter;
			// 
			// UpdateButton
			// 
			this.UpdateButton.BackColor = System.Drawing.SystemColors.Control;
			this.UpdateButton.Location = new System.Drawing.Point(224, 104);
			this.UpdateButton.Name = "UpdateButton";
			this.UpdateButton.TabIndex = 8;
			this.UpdateButton.Text = "Update";
			this.UpdateButton.Click += new System.EventHandler(this.UpdateButton_Click);
			// 
			// frmTimerOutput
			// 
			this.AutoScaleBaseSize = new System.Drawing.Size(6, 13);
			this.BackColor = System.Drawing.SystemColors.Window;
			this.ClientSize = new System.Drawing.Size(332, 251);
			this.Controls.Add(this.UpdateButton);
			this.Controls.Add(this.cmdEndProgram);
			this.Controls.Add(this.txtFrequencyToSet);
			this.Controls.Add(this.lblShowFrequency);
			this.Controls.Add(this.lblFrequency);
			this.Controls.Add(this.lblShowValue);
			this.Controls.Add(this.lblValueSent);
			this.Controls.Add(this.lblUseScroll);
			this.Controls.Add(this.lblDemoFunction);
			this.Font = new System.Drawing.Font("Arial", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.ForeColor = System.Drawing.SystemColors.WindowText;
			this.Location = new System.Drawing.Point(7, 103);
			this.Name = "frmTimerOutput";
			this.StartPosition = System.Windows.Forms.FormStartPosition.Manual;
			this.Text = "Universal Library Timer Output ";
			this.ResumeLayout(false);

		}

	#endregion

		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main() 
		{
			Application.Run(new frmTimerOutput());
		}


		private void cmdEndProgram_Click(object eventSender, System.EventArgs eventArgs) /* Handles cmdEndProgram.Click */
		{
			DaqBoard.TimerOutStop(0);
			Application.Exit();
		}


		private void UpdateButton_Click(object sender, System.EventArgs e) /* Handles UpdateButton.Click */
		{
			 bool IsValidNumber = true;
			 double frequency = 1000000.0;
			try
         {
				frequency = double.Parse(txtFrequencyToSet.Text);     
			}

        catch (Exception ex )
			{ 
				MessageBox.Show(txtFrequencyToSet.Text + " is not a valid frequency value", "Invalid Frequency ", MessageBoxButtons.OK, MessageBoxIcon.Error);
					IsValidNumber = false;
			}
        
			if (IsValidNumber)
			{
				double frequencySet = frequency;
				//  Parameters:
				//    TimerNum       :the timer output channel
				//    Frequency      :the frequency to output
				MccDaq.ErrorInfo ULStat = DaqBoard.TimerOutStart(0, ref frequency);
				if(ULStat.Value == MccDaq.ErrorInfo.ErrorCode.NoErrors)
				{
				  lblValueSent.Text = "The frequency sent to timer 0 was:";
				  lblFrequency.Text = "The frequency output from timer channel 0 is:";
				  lblShowValue.Text = frequencySet.ToString("0.0##") + " Hz";
				  lblShowFrequency.Text = frequency.ToString("0.0##") + " Hz";
				}
			}
		}

	}
}

File: measurecompute/DAQ/VB.NET/TimerOutStart01.NET/TimerOutStart01.vb
'==============================================================================

' File:                         TimerOut01.VB

'  Library Call Demonstrated:    MccDaq.MccBoard.TimerOutStart()
'								 MccDaq.MccBoard.TimerOutStop()

'  Purpose:                      Controls an Output Timer Channel.

'  Demonstration:                Sends a frequency output to Timer 0.

'  Other Library Calls:          MccDaq.MccService.ErrHandling()

'  Special Requirements:         Board 0 must have a Timer output.

'==============================================================================
Option Strict Off
Option Explicit On 
Public Class frmTimerOutput
  Inherits System.Windows.Forms.Form


    Private DaqBoard As MccDaq.MccBoard = New MccDaq.MccBoard(0)

  Private Sub UpdateButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles UpdateButton.Click
    Dim ULStat As MccDaq.ErrorInfo
    Dim Frequency As Double
    Dim IsValidNumber As Boolean = True

    lblShowValue.Text = ""
    lblShowFrequency.Text = ""

    Try
      Frequency = Double.Parse(txtFrequencyToSet.Text)
    Catch ex As Exception
      MessageBox.Show(txtFrequencyToSet.Text + " is not a valid frequency value", "Invalid Frequency ", MessageBoxButtons.OK, MessageBoxIcon.Error)
      IsValidNumber = False
    End Try

    If (IsValidNumber) Then
      Dim FrequencySet As Double = Frequency
      ' Parameters:
      '   TimerNum       :the timer output channel
      '   Frequency      :the frequency to output
      ULStat = DaqBoard.TimerOutStart(0, Frequency)
      If ULStat.Value <> MccDaq.ErrorInfo.ErrorCode.NoErrors Then Stop

      lblValueSent.Text = "The frequency sent to timer 0 was:"
      lblFrequency.Text = "The frequency output from timer channel 0 is:"
      lblShowValue.Text = FrequencySet.ToString("0.0##") + " Hz"
      lblShowFrequency.Text = Frequency.ToString("0.0##") + " Hz"
    End If

  End Sub

#Region "Windows Form Designer generated code "
  Public Sub New()
    MyBase.New()

    'This call is required by the Windows Form Designer.
    InitializeComponent()

    InitUL()

  End Sub
  'Form overrides dispose to clean up the component list.
  Protected Overloads Overrides Sub Dispose(ByVal Disposing As Boolean)
    If Disposing Then
      If Not components Is Nothing Then
        components.Dispose()
      End If
    End If
    MyBase.Dispose(Disposing)
  End Sub
  'Required by the Windows Form Designer
  Private components As System.ComponentModel.IContainer
  Public ToolTip1 As System.Windows.Forms.ToolTip
  Public WithEvents cmdEndProgram As System.Windows.Forms.Button
  Public WithEvents lblShowValue As System.Windows.Forms.Label
  Public WithEvents lblValueSent As System.Windows.Forms.Label
  Public WithEvents lblUseScroll As System.Windows.Forms.Label
  Public WithEvents lblDemoFunction As System.Windows.Forms.Label
  'NOTE: The following procedure is required by the Windows Form Designer
  'It can be modified using the Windows Form Designer.
  'Do not modify it using the code editor.
  Friend WithEvents UpdateButton As System.Windows.Forms.Button
  Public WithEvents lblShowFrequency As System.Windows.Forms.Label
  Public WithEvents lblFrequency As System.Windows.Forms.Label
  Public WithEvents txtFrequencyToSet As System.Windows.Forms.TextBox
  <System.Diagnostics.DebuggerStepThrough()> Private Sub InitializeComponent()
    Me.components = New System.ComponentModel.Container
    Me.ToolTip1 = New System.Windows.Forms.ToolTip(Me.components)
    Me.cmdEndProgram = New System.Windows.Forms.Button
    Me.txtFrequencyToSet = New System.Windows.Forms.TextBox
    Me.lblShowFrequency = New System.Windows.Forms.Label
    Me.lblFrequency = New System.Windows.Forms.Label
    Me.lblShowValue = New System.Windows.Forms.Label
    Me.lblValueSent = New System.Windows.Forms.Label
    Me.lblUseScroll = New System.Windows.Forms.Label
    Me.lblDemoFunction = New System.Windows.Forms.Label
    Me.UpdateButton = New System.Windows.Forms.Button
    Me.SuspendLayout()
    '
    'cmdEndProgram
    '
    Me.cmdEndProgram.BackColor = System.Drawing.SystemColors.Control
    Me.cmdEndProgram.Cursor = System.Windows.Forms.Cursors.Default
    Me.cmdEndProgram.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
    Me.cmdEndProgram.ForeColor = System.Drawing.SystemColors.ControlText
    Me.cmdEndProgram.Location = New System.Drawing.Point(248, 216)
    Me.cmdEndProgram.Name = "cmdEndProgram"
    Me.cmdEndProgram.RightToLeft = System.Windows.Forms.RightToLeft.No
    Me.cmdEndProgram.Size = New System.Drawing.Size(55, 26)
    Me.cmdEndProgram.TabIndex = 5
    Me.cmdEndProgram.Text = "Quit"
    '
    'txtFrequencyToSet
    '
    Me.txtFrequencyToSet.AcceptsReturn = True
    Me.txtFrequencyToSet.AutoSize = False
    Me.txtFrequencyToSet.BackColor = System.Drawing.SystemColors.Window
    Me.txtFrequencyToSet.BorderStyle = System.Windows.Forms.BorderStyle.FixedSingle
    Me.txtFrequencyToSet.Cursor = System.Windows.Forms.Cursors.IBeam
    Me.txtFrequencyToSet.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
    Me.txtFrequencyToSet.ForeColor = System.Drawing.SystemColors.WindowText
    Me.txtFrequencyToSet.Location = New System.Drawing.Point(120, 104)
    Me.txtFrequencyToSet.MaxLength = 0
    Me.txtFrequencyToSet.Name = "txtFrequencyToSet"
    Me.txtFrequencyToSet.RightToLeft = System.Windows.Forms.RightToLeft.No
    Me.txtFrequencyToSet.Size = New System.Drawing.Size(81, 25)
    Me.txtFrequencyToSet.TabIndex = 0
    Me.txtFrequencyToSet.Text = "100000"
    '
    'lblShowFrequency
    '
    Me.lblShowFrequency.BackColor = System.Drawing.SystemColors.Window
    Me.lblShowFrequency.Cursor = System.Windows.Forms.Cursors.Default
    Me.lblShowFrequency.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
    Me.lblShowFrequency.ForeColor = System.Drawing.Color.Blue
    Me.lblShowFrequency.Location = New System.Drawing.Point(248, 176)
    Me.lblShowFrequency.Name = "lblShowFrequency"
    Me.lblShowFrequency.RightToLeft = System.Windows.Forms.RightToLeft.No
    Me.lblShowFrequency.Size = New System.Drawing.Size(80, 17)
    Me.lblShowFrequency.TabIndex = 6
    '
    'lblFrequency
    '
    Me.lblFrequency.BackColor = System.Drawing.SystemColors.Window
    Me.lblFrequency.Cursor = System.Windows.Forms.Cursors.Default
    Me.lblFrequency.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
    Me.lblFrequency.ForeColor = System.Drawing.SystemColors.WindowText
    Me.lblFrequency.Location = New System.Drawing.Point(8, 176)
    Me.lblFrequency.Name = "lblFrequency"
    Me.lblFrequency.RightToLeft = System.Windows.Forms.RightToLeft.No
    Me.lblFrequency.Size = New System.Drawing.Size(240, 17)
    Me.lblFrequency.TabIndex = 7
    '
    'lblShowValue
    '
    Me.lblShowValue.BackColor = System.Drawing.SystemColors.Window
    Me.lblShowValue.Cursor = System.Windows.Forms.Cursors.Default
    Me.lblShowValue.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
    Me.lblShowValue.ForeColor = System.Drawing.Color.Blue
    Me.lblShowValue.Location = New System.Drawing.Point(248, 159)
    Me.lblShowValue.Name = "lblShowValue"
    Me.lblShowValue.RightToLeft = System.Windows.Forms.RightToLeft.No
    Me.lblShowValue.Size = New System.Drawing.Size(80, 17)
    Me.lblShowValue.TabIndex = 4
    '
    'lblValueSent
    '
    Me.lblValueSent.BackColor = System.Drawing.SystemColors.Window
    Me.lblValueSent.Cursor = System.Windows.Forms.Cursors.Default
    Me.lblValueSent.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
    Me.lblValueSent.ForeColor = System.Drawing.SystemColors.WindowText
    Me.lblValueSent.Location = New System.Drawing.Point(8, 159)
    Me.lblValueSent.Name = "lblValueSent"
    Me.lblValueSent.RightToLeft = System.Windows.Forms.RightToLeft.No
    Me.lblValueSent.Size = New System.Drawing.Size(240, 17)
    Me.lblValueSent.TabIndex = 3
    '
    'lblUseScroll
    '
    Me.lblUseScroll.BackColor = System.Drawing.SystemColors.Window
    Me.lblUseScroll.Cursor = System.Windows.Forms.Cursors.Default
    Me.lblUseScroll.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
    Me.lblUseScroll.ForeColor = System.Drawing.SystemColors.WindowText
    Me.lblUseScroll.Location = New System.Drawing.Point(56, 64)
    Me.lblUseScroll.Name = "lblUseScroll"
    Me.lblUseScroll.RightToLeft = System.Windows.Forms.RightToLeft.No
    Me.lblUseScroll.Size = New System.Drawing.Size(225, 32)
    Me.lblUseScroll.TabIndex = 2
    Me.lblUseScroll.Text = "Enter a frequency within the timer's range and click Update Button when done"
    '
    'lblDemoFunction
    '
    Me.lblDemoFunction.BackColor = System.Drawing.SystemColors.Window
    Me.lblDemoFunction.Cursor = System.Windows.Forms.Cursors.Default
    Me.lblDemoFunction.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
    Me.lblDemoFunction.ForeColor = System.Drawing.SystemColors.WindowText
    Me.lblDemoFunction.Location = New System.Drawing.Point(38, 24)
    Me.lblDemoFunction.Name = "lblDemoFunction"
    Me.lblDemoFunction.RightToLeft = System.Windows.Forms.RightToLeft.No
    Me.lblDemoFunction.Size = New System.Drawing.Size(256, 25)
    Me.lblDemoFunction.TabIndex = 1
    Me.lblDemoFunction.Text = "Demonstration of MccDaq.MccBoard.TimerStart() and MccDaq.MccBoard.TimerStop()"
    Me.lblDemoFunction.TextAlign = System.Drawing.ContentAlignment.TopCenter
    '
    'UpdateButton
    '
    Me.UpdateButton.BackColor = System.Drawing.SystemColors.Control
    Me.UpdateButton.Location = New System.Drawing.Point(224, 104)
    Me.UpdateButton.Name = "UpdateButton"
    Me.UpdateButton.TabIndex = 8
    Me.UpdateButton.Text = "Update"
    '
    'frmTimerOutput
    '
    Me.AutoScaleBaseSize = New System.Drawing.Size(6, 13)
    Me.BackColor = System.Drawing.SystemColors.Window
    Me.ClientSize = New System.Drawing.Size(332, 251)
    Me.Controls.Add(Me.UpdateButton)
    Me.Controls.Add(Me.cmdEndProgram)
    Me.Controls.Add(Me.txtFrequencyToSet)
    Me.Controls.Add(Me.lblShowFrequency)
    Me.Controls.Add(Me.lblFrequency)
    Me.Controls.Add(Me.lblShowValue)
    Me.Controls.Add(Me.lblValueSent)
    Me.Controls.Add(Me.lblUseScroll)
    Me.Controls.Add(Me.lblDemoFunction)
    Me.Font = New System.Drawing.Font("Arial", 8.25!, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
    Me.ForeColor = System.Drawing.SystemColors.WindowText
    Me.Location = New System.Drawing.Point(7, 103)
    Me.Name = "frmTimerOutput"
    Me.StartPosition = System.Windows.Forms.FormStartPosition.Manual
    Me.Text = "Universal Library Timer Output "
    Me.ResumeLayout(False)

  End Sub
#End Region

#Region "Universal Library Initialization - Expand this region to change error handling, etc."

  Private Sub InitUL()
    Dim ULStat As MccDaq.ErrorInfo

    ' declare revision level of Universal Library


    ULStat = MccDaq.MccService.DeclareRevision(MccDaq.MccService.CurrentRevNum)

    ' Initiate error handling
    '  activating error handling will trap errors like
    '  bad channel numbers and non-configured conditions.
    '  Parameters:
    '    MccDaq.ErrorReporting.PrintAll :all warnings and errors encountered will be printed
    '    MccDaq.ErrorHandling.StopAll   :if any error is encountered, the program will stop

    ULStat = MccDaq.MccService.ErrHandling(MccDaq.ErrorReporting.PrintAll, MccDaq.ErrorHandling.StopAll)
    If ULStat.Value <> MccDaq.ErrorInfo.ErrorCode.NoErrors Then
      Stop
    End If


  End Sub
#End Region

  Private Sub cmdEndProgram_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmdEndProgram.Click
    DaqBoard.TimerOutStop(0)
    Me.Close()
  End Sub
End Class